rootfs/etc/nginx/template/nginx.tmpl

```diff
@@ -358,7 +358,7 @@ http {
     {{ range $name, $upstream := $backends }}
     {{ if eq $upstream.SessionAffinity.AffinityType "cookie" }}
     upstream sticky-{{ $upstream.Name }} {
-        sticky hash={{ $upstream.SessionAffinity.CookieSessionAffinity.Hash }} name={{ $upstream.SessionAffinity.CookieSessionAffinity.Name }}  httponly;
+        sticky hash={{ $upstream.SessionAffinity.CookieSessionAffinity.Hash }} name={{ $upstream.SessionAffinity.CookieSessionAffinity.Name }}{{if eq (len $upstream.SessionAffinity.CookieSessionAffinity.Locations) 1 }}{{ range $locationName, $locationPaths := $upstream.SessionAffinity.CookieSessionAffinity.Locations }}{{ if eq (len $locationPaths) 1 }} path={{ index $locationPaths 0 }}{{ end }}{{ end }}{{ end }} httponly;
```

Comment:
As a bonus, you could wrap this logic inside a template function (see `internal/ingress/controller/template/template.go`) that receives a `*ingress.CookieSessionAffinity` and returns either `path=/xyz; ` or nothing.
You can then call that function inside the template like `{{ cookiepath $upstream.SessionAffinity.CookieSessionAffinity }}`.

No hard feeling on my side though, the Go template language also does the trick here.